pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -270,18 +271,7 @@ func (c *Controller) Cluster() string {
 	return c.clusterID
 }
 
-func (c *Controller) checkReadyForEvents() error {
-	if !c.HasSynced() {
-		return errors.New("waiting till full synchronization")
-	}
-	return nil
-}
-
 func (c *Controller) onServiceEvent(curr interface{}, event model.Event) error {
-	if err := c.checkReadyForEvents(); err != nil {
```

Comment:
why do we not need this?

Reply:
Before queue run, we have waited cache sync. That is same as what `checkReadyForEvents` does